mrp_production_product_category_restriction/models/mrp_production.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    def action_confirm(self):
+        category_ids = self.mapped("product_id.categ_id")
+        categories_not_produce = category_ids.filtered(lambda c: not c.produce_ok)
+        if categories_not_produce:
+            mos_in_categories_not_produce = self.filtered(
+                lambda mo: mo.product_id.categ_id in categories_not_produce
+            )
+            mo_names = mos_in_categories_not_produce.mapped("name")
+            mo_names_str = ", ".join(mo_names)
+            warning_message = (
+                "The following manufacturing orders are for products in categories "
+                "not allowed to produce: %s" % mo_names_str
+            )
+            raise UserError(_(warning_message))
```

Comment:
I think this can be simplified.

```suggestion
        for production in self:
            product = production.product_id
            if not product.categ_id.produce_ok:
                raise UserError(_("%s is not allowed to be produced according to the product category setting." % product.name))
```